### PR TITLE
Bug 1877123 - Fix number localization in `TabCounter` and `TabTools`

### PR DESCRIPTION
**Pull request by MozillaNoah**
| Before | After |
| -- | -- |
| image |  image |

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1877123

### CODE REVIEW

File: fenix/app/src/main/java/org/mozilla/fenix/compose/ext/Int.kt [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+package org.mozilla.fenix.compose.ext
+
+import androidx.compose.ui.text.intl.Locale
+import java.text.NumberFormat
+import java.util.Locale as JavaLocale
+
+/**
+ * Returns a localized string representation of the value.
+ */
+fun Int.toLocaleString(): String =
```
Comment (MozillaNoah) [inline fenix/app/src/main/java/org/mozilla/fenix/compose/ext/Int.kt:14]: I decided to create a helper function for this.

I needed to use `Locale.current.language` in order to get the Compose previews working. For whatever reason, Java's `Locale.default()` does not work correctly in previews, causing the Locale preview parameter to be ignored when formatting numbers.

Comment (gabrielluong) [inline fenix/app/src/main/java/org/mozilla/fenix/compose/ext/Int.kt:14]: Is it possible to add a test for this?

**Review by gabrielluong — APPROVED** [review]

File: fenix/app/src/main/java/org/mozilla/fenix/compose/ext/Int.kt (hunk 1 above)
Comment (MozillaNoah) [reply fenix/app/src/main/java/org/mozilla/fenix/compose/ext/Int.kt:14]: Not sure; I'll see how testable Locales are

Comment (gabrielluong) [reply fenix/app/src/main/java/org/mozilla/fenix/compose/ext/Int.kt:14]: Noice :shipit:

File: fenix/app/src/test/java/org/mozilla/fenix/compose/ext/IntTest.kt [hunk 2]
```diff
@@ -0,0 +1,23 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.compose.ext
+
+import org.junit.Assert.assertEquals
+import java.util.Locale as JavaLocale
+import org.junit.Test
```
Comment (gabrielluong) [inline fenix/app/src/test/java/org/mozilla/fenix/compose/ext/IntTest.kt:9]: I think we need to reorder the imports before one of the linters complain?

**Pull request merged**